Cargo.toml

```diff
@@ -12,6 +12,7 @@ categories = ["api-bindings", "development-tools::ffi"]
 license = "Apache-2.0"
 exclude = ["/.gitignore", ".cargo/config", "/codecov.yml", "/Makefile", "/pyproject.toml", "/noxfile.py"]
 edition = "2018"
+rust-version = "1.48"
```

Comment:
Unfortunately this produces a warning on rust < 1.56: https://github.com/PyO3/pyo3/runs/6972028061?check_suite_focus=true#step:13:18 , so this won't be useful until we actually raise msrv above that :(

Reply:
I don't think that warning is a reason to revert this as it is limited to building PyO3 itself and will not bother dependencies using the MSRV. And it is useful machine-readable and highly discoverable human-readable metadata.

Reply:
Yeah I'm torn on this one. Can we think if there any cases where the warning might lead to build errors?

Reply:
I am not aware of situations where warnings in dependencies would be upgraded to errors. At least when building using Cargo and going out of one's way by e.g. grepping build logs instead of using Cargo/rustc flags to deny warnings.

Reply:
(Just as another data point to make this more palatable: `rust-numpy` has this since 0.16.0 with no reports so far.)

Reply:
Thanks - either way I agree I will revert from this PR and make a separate one. I wasn't trying to sneak it in 😛.